lib/isNotNullish.ts

```diff
@@ -1,5 +1,5 @@
-export function isNotNullish<T>(
-  value: T
-): value is Exclude<T, null | undefined> {
+export type NotNullish<T> = Exclude<T, null | undefined>;
```

Comment:
suggestion: let's call this `NonNullish` to align with `NonEmptyArray` (both should be using `non` instead of `not` as prefix)

Reply:
Good shout, updating the PR.